[opam 2.1~beta4] ./configure requires an OCaml compiler but a vendored OCaml compiler will be compiled anyway

Here is my setup:
```
$ cat /etc/os-release | grep ID=
ID=ubuntu
VERSION_ID="18.04"
$ git clone git://github.com/ocaml/opam.git
$ cd opam
$ ./configure
./configure
checking for ocamlc... no
checking for ocaml... no
checking for ocamldep... no
checking for ocamlmktop... no
checking for ocamlmklib... no
checking for ocamldoc... no
checking for ocamlbuild... no
configure: error: You must install the OCaml compiler
$ sudo apt install ocaml-nox
$ ocaml -vnum
4.05.0
$ ./configure
$ make lib-ext
$ make
[...]
make[4]: Entering directory '/home/kit_ty_kate/opam/src_ext/secondary/ocaml-4.11.1/stdlib'
../boot/ocamlrun ../ocamlc -strict-sequence -absname -w +a-4-9-41-42-44-45-48 -g -warn-error A -bin-annot -nostdlib -safe-string -strict-formats  -nopervasives -c camlinternalFormatBasics.mli
../boot/ocamlrun ../ocamlc -strict-sequence -absname -w +a-4-9-41-42-44-45-48 -g -warn-error A -bin-annot -nostdlib -safe-string -strict-formats  -nopervasives -c camlinternalFormatBasics.ml
```
The last bit indicates that a vendored OCaml compiler is being compiled. However the configure script demands to have an OCaml compiler installed. To me, this seems like extra steps that's not needed.

--- a/.gitattributes
+++ b/.gitattributes
@@ -3,7 +3,7 @@
 
 # Shell scripts, autoconf, etc. must have LF endings, even on Windows
 *.sh text eol=lf
-configure text eol=lf -diff
+configure text eol=lf -diff linguist-generated
 configure.ac text eol=lf
 msvs-detect text eol=lf
 check_linker text eol=lf

--- a/configure
+++ b/configure
@@ -2177,6 +2177,9 @@ if  test -x bootstrap/ocaml/bin/ocamlc -o -x bootstrap/ocaml/bin/ocamlopt ; then
 
 fi
 
+# XXX This isn't strictly correct for Windows
+MIN_OCAML_VERSION=4.02.3
+
   # checking for ocamlc
   if test -n "$ac_tool_prefix"; then
   # Extract the first word of "${ac_tool_prefix}ocamlc", so it can be a program name with args.
@@ -3386,7 +3389,7 @@ fi
 
 
 if test "x$OCAMLC" = "xno"; then
-   as_fn_error $? "You must install the OCaml compiler (or run: make compiler)" "$LINENO" 5
+   as_fn_error $? "You must install the OCaml compiler, at least version $MIN_OCAML_VERSION (or run: make compiler)" "$LINENO" 5
 fi
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking OCaml Sys.os_type" >&5
 $as_echo_n "checking OCaml Sys.os_type... " >&6; }
@@ -3535,8 +3538,6 @@ if test "x${enable_version_check}" != "xno"; then :
 
   if  test ${WIN32} -eq 1 ; then :
   MIN_OCAML_VERSION=4.06.0
-else
-  MIN_OCAML_VERSION=4.02.3
 fi
 
 

--- a/configure
+++ b/configure
@@ -3386,7 +3386,7 @@ fi
 
 
 if test "x$OCAMLC" = "xno"; then
-   as_fn_error $? "You must install the OCaml compiler" "$LINENO" 5
+   as_fn_error $? "You must install the OCaml compiler (or run: make compiler)" "$LINENO" 5
 fi
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking OCaml Sys.os_type" >&5
 $as_echo_n "checking OCaml Sys.os_type... " >&6; }

--- a/configure.ac
+++ b/configure.ac
@@ -16,7 +16,7 @@ AS_IF([ test -x bootstrap/ocaml/bin/ocamlc -o -x bootstrap/ocaml/bin/ocamlopt ],
 
 AC_PROG_OCAML
 if test "x$OCAMLC" = "xno"; then
-   AC_MSG_ERROR([You must install the OCaml compiler])
+   AC_MSG_ERROR([You must install the OCaml compiler (or run: make compiler)])
 fi
 AC_CHECK_OCAML_OS_TYPE
 AS_IF([ test "${OCAML_OS_TYPE}" = "Win32"],[

--- a/configure.ac
+++ b/configure.ac
@@ -14,9 +14,12 @@ AS_IF([ test -x bootstrap/ocaml/bin/ocamlc -o -x bootstrap/ocaml/bin/ocamlopt ],
   export PATH="$PATH_PREPEND$PATH"
 ])
 
+# XXX This isn't strictly correct for Windows
+MIN_OCAML_VERSION=4.02.3
+
 AC_PROG_OCAML
 if test "x$OCAMLC" = "xno"; then
-   AC_MSG_ERROR([You must install the OCaml compiler (or run: make compiler)])
+   AC_MSG_ERROR([You must install the OCaml compiler, at least version $MIN_OCAML_VERSION (or run: make compiler)])
 fi
 AC_CHECK_OCAML_OS_TYPE
 AS_IF([ test "${OCAML_OS_TYPE}" = "Win32"],[
@@ -81,7 +84,7 @@ AC_SUBST(LIBRARY_PATH,$LIBRARY_PATH)
 # Check that OCaml version is greater or equal to 4.02.3
 # Native Windows builds require at least 4.06.0 for the Unicode runtime.
 AS_IF([test "x${enable_version_check}" != "xno"], [
-  AS_IF([ test ${WIN32} -eq 1 ],[MIN_OCAML_VERSION=4.06.0],[MIN_OCAML_VERSION=4.02.3])
+  AS_IF([ test ${WIN32} -eq 1 ],[MIN_OCAML_VERSION=4.06.0])
   AX_COMPARE_VERSION(
     [$OCAMLVERSION], [lt], [$MIN_OCAML_VERSION],
     AC_MSG_ERROR([Your version of OCaml: $OCAMLVERSION is not supported]))

--- a/master_changes.md
+++ b/master_changes.md
@@ -77,6 +77,7 @@ New option/command/subcommand are prefixed with ◈.
   * Makefile: Add rule `custom-libinstall` for `opam-custom-install` use [#4401 @AltGr]
   * opam-client lib: fix dependency constraints for cmdliner, extlib [#4410 @AltGr]
   * Makefile: Fix missing spaces between `DUNE_ARGS` and `DUNE_PROMOTE_ARG`. [#4458 @nbraud - fix #4457]
+  * advertise make compiler in configure if no compiler is found [#4451 @dra27]
 
 ## Infrastructure
   *

